### PR TITLE
importccl: ignore mysql's _binary introducer

### LINKED ISSUE #31169: mysqldump import: binary fields proceeded by `_binary` unsupported
[as of](https://github.com/mysql/mysql-server/commit/0ac7a57eaf56bbe8b3dbe2f5af0fa4f61480ba90) 5.7.23, mysqldump dumps binary fields with the `_binary` prefix:

> If mysqldump or mysqlpump were used to dump binary data without the --hex-blob option, reloading the dump file could produce spurious warnings (values were inserted correctly regardless of the warnings). Such values are now written preceded by the _binary introducer to silence the warnings. (Bug #22601255, Bug #80150)

This syntax is not currently supported by the mysql parser used by cockroach to part mysql import data, resulting in an error when dump files containing binary fields dumped by current mysqldump from 8.x and 5.x releases are used in `IMPORT`.

### PR DESCRIPTION
**Pull request by dt**
MySQL version 5.7.23 included a change to mysqldump, compared to .22, to prefix binary fields with the '_binary' introducer.
These are described in the mysql docs here: https://dev.mysql.com/doc/refman/5.7/en/charset-introducer.html

Fixes #31169.

Release note (sql change): handle binary fields dumped by muysqldump v5.7.23 with _binary prefix.

### CODE REVIEW

File: pkg/ccl/importccl/read_import_mysql.go [hunk 1]
```diff
@@ -183,25 +183,30 @@ func mysqlValueToDatum(
...
+			case *tree.DDecimal:
+				dec := &i.Decimal
+				dd := &tree.DDecimal{}
+				dd.Decimal.Neg(dec)
+				return dd, nil
+			default:
+				return nil, errors.Errorf("unsupported negation of %T", i)
+			}
+		case "_binary", "_binary ":
+			// TODO(dt): do something with this?
```
Comment (mjibson) [inline pkg/ccl/importccl/read_import_mysql.go]: this todo is kind of vague. can it have a bit more description?

**Review by madelynnblue — APPROVED** [review]

File: pkg/ccl/importccl/read_import_mysql.go (hunk 1 above)
Comment (dt) [reply pkg/ccl/importccl/read_import_mysql.go]: Done

**Comment by knz** [comment]
maybe this wants a backport

**Pull request merged**